free-2.md

```diff
@@ -11,41 +11,90 @@ _What need is met by your idea? Who are you helping? What is that person's
+that they can run and debug.
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+I liked the idea of doing a re-implementation project described in the homework,
+and I thought that this might be a unique project to apply the re-implementation
+technique to. I came up with it because I was thinking about different smaller
+programs that I might want to re-implement, thought of HMMM, read the
+description page, and thought it would be a good fit!
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Assembly language for CS5 learning.
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+The user would be able to code in the assembly language, then have it be
+assembled, and then run. There should also be a debugging interface, where the
+user can walk through each line, and the program will show what information is
+in the registers and the stack as each line is executed.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the assembly language will be executed, and the output will
+be shown to the user. Errors could include syntax errors, or jumps that go
+to lines of code that do not exist. These errors would be communicated to the
+user during compilation for synax errors, and at runtime if jumping to a line
+that does not exist.
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to write very simple assembly language programs, and it should
+be easy to have access to visualizing the registers and stack so that students
+can debug and understand the language. It should be difficult to create complex
+programs, we also do not have a compiler that writes in HMMM so it would be
+difficult to convert a program into HMMM.
+
 ### Related work
 
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+Yes, this DSL already does exist because this is a re-implementation project.
+https://www.cs.hmc.edu/~cs5grad/cs5/hmmm/documentation/documentation.html
+http://writehmmm-hmccsstudio.pythonanywhere.com/
+
+The language already does address the need very well, it already exists. I think
+there could be additional room for debugging options for students, but also I
+wouldn't want to make HMMM too easy for students to use by adding too many extra
+debugging techniques.
```

Comment:
I think you make a valid point here; you want to strike a good balance between simplifying it for students without giving the answer away. A good approach here might be to talk with the current profs for CS5 (or the iconic Dodds).